integration-tests/tests/src/tests/mixed.ts

```diff
@@ -839,78 +2714,180 @@ describe("Mixed", () => {
+            mixed: { key: "original" },
           });
           expectRealmDictionary(dictionary);
 
+          // Assign the embedded object to the collections.
           expect(() => (list[0] = embeddedObject)).to.throw(
             "Using an embedded object (EmbeddedObject) as a Mixed value is not supported",
           );
-          expect(
-            () => (dictionary.prop = embeddedObject),
+          expect(() => (dictionary.key = embeddedObject)).to.throw(
             "Using an embedded object (EmbeddedObject) as a Mixed value is not supported",
           );
         });
         const objects = this.realm.objects<IMixedAndEmbedded>(MixedAndEmbeddedSchema.name);
         expect(objects.length).equals(3);
-        // Check that the list and dictionary are still empty.
-        expect((objects[1].mixedValue as Realm.List<any>).length).equals(0);
-        expect(Object.keys(objects[2].mixedValue as Realm.Dictionary<any>).length).equals(0);
+
+        // Check that the list and dictionary are unchanged.
+        const list = objects[1].mixed;
+        expectRealmList(list);
+        expect(list[0]).equals("original");
+
+        const dictionary = objects[2].mixed;
+        expectRealmDictionary(dictionary);
+        expect(dictionary.key).equals("original");
       });
 
       it("throws when setting a list or dictionary outside a transaction", function (this: RealmContext) {
         const created = this.realm.write(() => {
-          return this.realm.create<IMixedSchema>(MixedSchema.name, { value: "original" });
+          return this.realm.create<IMixedSchema>(MixedSchema.name, { mixed: "original" });
         });
-        expect(created.value).equals("original");
-        expect(() => (created.value = ["a list item"])).to.throw(
+        expect(created.mixed).equals("original");
+        expect(() => (created.mixed = ["a list item"])).to.throw(
           "Cannot modify managed objects outside of a write transaction",
         );
-        expect(() => (created.value = { key: "a dictionary value" })).to.throw(
+        expect(() => (created.mixed = { key: "a dictionary value" })).to.throw(
           "Cannot modify managed objects outside of a write transaction",
         );
-        expect(created.value).equals("original");
+        expect(created.mixed).equals("original");
+      });
+
+      it("throws when setting a list item out of bounds", function (this: RealmContext) {
+        const { mixed: list } = this.realm.write(() => {
+          // Create an empty list as the Mixed value.
+          return this.realm.create<IMixedSchema>(MixedSchema.name, { mixed: [] });
+        });
+        expectRealmList(list);
+        expect(list.length).equals(0);
+
+        expect(() => {
+          this.realm.write(() => {
+            list[0] = "primitive";
+          });
+        }).to.throw("Requested index 0 calling set() on list 'MixedClass.mixed' when empty");
```

Comment:
I'm surprised this isn't allowed.

Reply:
This is where `Realm.List` differs in behavior from a JS Array. The Core APIs for *setting* will throw if `index >= size`, whereas *inserting* only throws for `index >= size + 1`. This is the same behavior as we had before for `Realm.List`.

From a JS perspective, setting a value at, say, index 5 of an empty list, we'd have to set index 0-4 to `undefined` (although this would be converted to `null` values).